src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1836,19 +1841,22 @@ public void serialize(RowFilter filter, DataOutputPlus out, int version) throws
         {
             out.writeBoolean(false); // Old "is for thrift" boolean
             FilterElement.serializer.serialize(filter.root, out, version);
+            out.writeBoolean(filter.allowFiltering);
         }
 
         public RowFilter deserialize(DataInputPlus in, int version, TableMetadata metadata) throws IOException
         {
             in.readBoolean(); // Unused
             FilterElement operation = FilterElement.serializer.deserialize(in, version, metadata);
-            return new RowFilter(operation);
+            boolean allowFiltering = in.readBoolean();
+            return new RowFilter(operation, allowFiltering);
         }
 
         public long serializedSize(RowFilter filter, int version)
         {
             return 1 // unused boolean
-                   + FilterElement.serializer.serializedSize(filter.root, version);
+                   + FilterElement.serializer.serializedSize(filter.root, version)
+                   + 1; // for allowFiltering
```

Comment:
This should depend on the messaging version. Also, it's probably better to use `TypeSizes.BOOL_SIZE`